pgx/src/subxact.rs

```diff
@@ -0,0 +1,196 @@
+use crate::{pg_sys, PgMemoryContexts, SpiClient};
+use std::fmt::{Debug, Formatter};
+use std::ops::Deref;
+
+/// Sub-transaction
+///
+/// Can be created by calling `SpiClient::sub_transaction`, `SubTransaction<Parent>::sub_transaction`
+/// or any other implementation of `SubTransactionExt` and obtaining it as an argument to the provided closure.
```

Comment:
My internal grammarian is trying to say the function names should be `subtransaction` (which is valid in English), whereas the type names should be `SubTransaction`, but that seems... somehow peculiar, given that normally those would normalize to either `sub_transaction` and `SubTransaction` like you have it here or `subtransaction` and `Subtransaction`.

Reply:
I named it `sub_transaction` for consistency with `SubTransaction`. But ultimately I don't have a strong opinion about this.

Reply:
And Postgres calls them "SubTransactions" in their API.

Reply:
Yeah, I think the type/trait names should use `SubTransaction`. Maybe I will start using `sub_transaction` in anger and it will bug me enough to want to take the `_` out, and I'll just let it be inconsistent-but-natural then. But otherwise it seems fine.